### PR TITLE
Tweak the behavior of search/replace bar

### LINKED ISSUE #19952: No replace in the ctrl + f search.
Godot version: 3.0.4 stable
OS/device including version: Win7

Issue description:
The replace feature seems to be missing when using ctrl + f. It's not intuitive to use ctrl + r, ctrl + r should just jump to the replace part of the ctrl + f search.
no search and replace

here is how it should be:
note

Search is in a small popup and there are many options to edit the searched text.

Steps to reproduce:
use the code editor

### LINKED ISSUE #20643: Pressing F3 in text editor when search field is closed makes permanent search highlights that you need CTRL + F to disable
Sooo... this issue requires a set of actions to appear, but paradoxically, it happened to me enough times to get me annoyed and I finally found a way to reproduce it.
tl;dr it causes a permanent search highlight to appear and you need to CTRL + F to cancel it.

**Steps to reproduce**

0. Here's example code to test it:
```
func _ready():
	var a = 1
	var b = 1
	var c = 1
```
1. Press CTRL + F
2. Type "1". This will highlight all 1s in your editor
image
3. Press ESC to close the search text field. It will correctly remove highlights
image
4. Press F3. The highlights are back, but there's no search field
image
You can't remove the highlights now. Pressing ESC does nothing, because there's no search field to close. You need to press CTRL + F to re-enable search field and then press ESC.

My suggested fixes:
-make F3 bring back search field if it's closed
OR
-make ESC always cancel search highlights even if search field is invisible

### PR DESCRIPTION
**Pull request by KoBeWi**
This PR brings some changes to how search/replace bar in script editor behaves, inspired by VS Code. This also closes #20643 and probably resolves #19952

The changes are:
- pressing Ctrl + F when search bar is invisible will no longer go to next result
- pressing (Shift +) F3 will now always open search bar
- text in replace bar isn't removed when it's opened
- pressing Ctrl + R with selected text or find bar visible will automatically focus and select replace bar (it doesn't happen when search is hidden and nothing is selected)
- search next/previous will skip the selected word even when the search bar is still invisible

There's one more thing that could be done (auto-fetching the word under cursor on F3 when search text is empty), but I'll leave that for later.

### CODE REVIEW

**Comment by KoBeWi** [comment]
Rebased with changes from #31239

**Comment by akien-mga** [comment]
Thanks!

**Comment by toger5** [comment]
Oh nice thank you!
